doc/source/whatsnew/v2.3.3.rst

```diff
@@ -25,6 +25,8 @@ Bug fixes
 - Fix bug in :meth:`Series.str.replace` using named capture groups (e.g., ``\g<name>``) with the Arrow-backed dtype would raise an error (:issue:`57636`)
 - Fix regression in ``~Series.str.contains``, ``~Series.str.match`` and ``~Series.str.fullmatch``
   with a compiled regex and custom flags (:issue:`62240`)
+- Fix regression in ``on_bad_lines`` callable when returning too many fields: now emits
```

Comment:
For pandas, we reserve "regression" for "it used to work, but now it doesn't". I don't believe that's the case here, is that right? If so, this should go in 3.0 instead of 2.3.3 and start out `Fix bug`. We generally only fix regressions in patch releases.